src/extended-keys.ts

```diff
@@ -55,29 +55,37 @@ export const decodeExtendedKey = (extendedKey: string): ExtendedKey => {
     );
   }
 
-  const version = buffer.readUInt32BE(0);
-  const depth = buffer.readUInt8(4);
+  const view = new DataView(
+    buffer.buffer,
+    buffer.byteOffset,
+    buffer.byteLength,
+  );
+
+  const version = view.getUint32(0, false);
+  const depth = view.getUint8(4);
   validateBIP44Depth(depth);
 
-  const parentFingerprint = buffer.readUInt32BE(5);
-  const index = buffer.readUInt32BE(9);
+  const parentFingerprint = view.getUint32(5, false);
+  const index = view.getUint32(9, false);
 
-  const chainCode = buffer.slice(13, 45);
+  const chainCode = buffer.subarray(13, 45);
   if (!isValidBufferKey(chainCode, 32)) {
     throw new Error(
       `Invalid extended key: Chain code must be a 32-byte non-zero Buffer.`,
     );
   }
 
-  const key = buffer.slice(45, 78);
+  const key = buffer.subarray(45, 78);
   if (!isValidBufferKey(key, 33)) {
     throw new Error(
       `Invalid extended key: Key must be a 33-byte non-zero Buffer.`,
     );
   }
 
+  const keyView = new DataView(key.buffer, key.byteOffset, key.byteLength);
+
   if (version === PUBLIC_KEY_VERSION) {
-    if (key.readUInt8(0) !== 0x02 && key.readUInt8(0) !== 0x03) {
+    if (keyView.getUint8(0) !== 0x02 && keyView.getUint8(0) !== 0x03) {
       throw new Error(
         `Invalid extended key: Public key must start with 0x02 or 0x03.`,
       );
```

Comment:
Why do we have a leading zero in the hex numbers here and below?

Reply:
While it would technically be the same, it's common to pad a hexadecimal string with an extra `0` when working with bytes. A single hexadecimal can be used to represent a nibble (4 bits), while two hexadecimal characters can be used to represent a byte. JavaScript interprets `0x02` and `0x03` as `2` and `3` regardless of the number of leading zeroes.